core/Plugin/Visualization.php

```diff
@@ -206,15 +207,17 @@ public function render()
             Log::error($logMessage);
 
             $loadingError = array('message' => $message);
+
+            $caughtException = $e;
```

Comment:
Actually I wonder if we would need some UI tests for each visualization when there is no data to see if any errors occur and whether all visualisations still render nicely? Eg if some view template is expecting a dataTable to be set when it possibly isn't it may fail... not sure if that is easy doable though and might be bit "painful" to write a test for.